### PR TITLE
Fix #2152: End server world tick too early and conditional

### LINKED ISSUE #2152: end world tick event too early and conditional
The mixin seems to be unfortunately placed in 1.18 and potentially other versions, causing the the event to not run if there are no players (contrary to the start world tick). It also runs too early, before entityManager.tick.

### PR DESCRIPTION
**Pull request by Technici4n**
Fix #2152

### CODE REVIEW

File: fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/client/ClientWorldMixin.java [hunk 1]
```diff
@@ -26,19 +26,17 @@
...
-		ClientTickEvents.END_WORLD_TICK.invoker().onEndTick((ClientWorld) (Object) this);
-	}
-
+public abstract class ClientWorldMixin {
 	@Inject(method = "tickEntities", at = @At("HEAD"))
 	private void startWorldTick(CallbackInfo ci) {
 		ClientTickEvents.START_WORLD_TICK.invoker().onStartTick((ClientWorld) (Object) this);
 	}
+
+	@Inject(method = "tickEntities", at = @At("TAIL"))
```
Comment (modmuss50) [inline fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/client/ClientWorldMixin.java:38]: Very similar to the 1.19 changes I had to make becuase of splitting. Seems to only differ by this injection target now: https://github.com/FabricMC/fabric/blob/1.19/fabric-lifecycle-events-v1/src/client/java/net/fabricmc/fabric/mixin/event/lifecycle/client/ClientWorldMixin.java#L38 👍

**Review by modmuss50 — APPROVED** [review]

File: fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/client/ClientWorldMixin.java (hunk 1 above)
Comment (Technici4n) [reply fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/client/ClientWorldMixin.java:38]: Isn't it the exact same inject, just placed differently in the source file and with a different method name?

Comment (Technici4n) [reply fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/client/ClientWorldMixin.java:38]: Seems like you fixed the issue in 1.19 for the server event, I didn't notice! 😮

Comment (modmuss50) [reply fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/client/ClientWorldMixin.java:38]: If I fixed it it was an accident 😂

Comment (modmuss50) [reply fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/client/ClientWorldMixin.java:38]: Ah yeah I did accidentally fix it 👍 Was confused as the method order is swapped.

**Pull request merged**